src/routing/approved.rs

```diff
@@ -1778,8 +1788,20 @@ impl Approved {
                 commands.extend(self.send_sync(self.section.clone(), self.network.clone())?);
             }
 
-            let sibling_key = if new_prefix != old_prefix {
-                self.section_key(&new_prefix.sibling()).copied()
+            let sibling_elders = if new_prefix != old_prefix {
+                if let Some(sibling_key) = self.section_key(&new_prefix.sibling()).copied() {
+                    if let Some(info) = self.network.get(&new_prefix.sibling()).cloned() {
```

Comment:
no need to use `cloned` ? 
you can use `let Some(ref info) = self.network.get(&new_prefix.sibling())` ?

same to the above `Some(sibling_key)` . only clone when assigning?